ngmt/modules/td/_pham.py

```diff
@@ -0,0 +1,372 @@
+        self.tracked_points = tracked_points
+        self.detected_turns = None
+
+    def detect(
+        self, data: pd.DataFrame, acceleration_data_unit: str, gyro_data_unit: str, sampling_freq_Hz: float, plot_results: bool = False
+    ) -> pd.DataFrame:
+        """
+        Detects truns based on the input accelerometer and gyro data.
+
+        Args:
+            data (pd.DataFrame): Input accelerometer and gyro data (N, 6) for x, y, and z axes.
+            acceleration_data_unit (str): Unit of acceleration data.
+            gyro_data_unit (str): Unit of gyro data.
+            sampling_freq_Hz (float): Sampling frequency of the input data.
+            plot_results (bool, optional): If True, generates a plot. Default is False.
+
+        Returns:
+            The turns information is stored in the 'detected_turns' attribute,
+            which is a pandas DataFrame in BIDS format with the following information:
+                - onset: Start time of the turn [s].
+                - duration: Duration of the turn [s].
+                - event_type: Type of the event (turn).
+                - direction_of_turn: Direction of turn which is either left or right.
+                - angle_of_turn_deg: Angle of the turn in degrees.
+                - peak_angular_velocity: Peak angular velocity [deg/s].
+                - tracking_systems: Tracking systems used (default is 'imu').
+                - tracked_points: Tracked points on the body (default is 'LowerBack').
+        """
+        # Check if data is a DataFrame
+        if not isinstance(data, pd.DataFrame):
+            raise ValueError("Input data must be a pandas DataFrame")
+
+        # Check if data has the correct shape
+        if data.shape[1] != 6:
+            raise ValueError("Input data must have 6 columns (3 for accelerometer and 3 for gyro)")
+
+        # Check if sampling frequency is positive
+        if sampling_freq_Hz <= 0:
+            raise ValueError("Sampling frequency must be positive")
+
+        # Check if plot_results is a boolean
+        if not isinstance(plot_results, bool):
+            raise ValueError("plot_results must be a boolean value")
+
+        # Select acceleration data and convert it to numpy array format
+        accel = data.iloc[:, 0:3].copy()
+        accel = accel.to_numpy()
+
+        # Select gyro data and convert it to numpy array format
+        gyro = data.iloc[:, 3:6].copy()
+        gyro = gyro.to_numpy()
+
+        # Check unit of acceleration data if it is in g or m/s^2
+        if acceleration_data_unit == "m/s^2":
+            # Convert acceleration data from m/s^2 to g (if not already is in g)
+            accel /= 9.81
+
+        # Check unit of acceleration data if it is in g or m/s^2
+        if gyro_data_unit == "rad/s":
+            # Convert acceleration data from rad/s to deg/s (if not already is in deg/s)
+            gyro = np.rad2deg(gyro)
+
+        # Convert gyro data unit from deg/s to rad/s
```

Comment:
Why do you check above whether data is in rad/s, and if so, convert it to deg/s, only to convert it back to rad/s here?

Reply:
I eliminated the conversion here. Now, it only checks once if it is in deg/s or not.